judge/judgedaemon.main.php

```diff
@@ -133,32 +138,49 @@ function request(string $url, string $verb = 'GET', $data = '', bool $failonerro
         curl_setopt($curl_handle, CURLOPT_POSTFIELDS, null);
     }
 
-    $response = curl_exec($curl_handle);
-    if ($response === false) {
-        $errstr = "Error while executing curl $verb to url " . $url . ": " . curl_error($curl_handle);
-        if ($failonerror) {
-            error($errstr);
+    $delay_in_ms = 1000;
+    $succeeded = false;
+    $response = null;
+    $errstr = null;
+
+    for ($trial = 1; $trial <= $steps; $trial++) {
+        $response = curl_exec($curl_handle);
+        if ($response === false) {
+            $errstr = "Error while executing curl $verb to url " . $url . ": " . curl_error($curl_handle);
         } else {
-            warning($errstr);
-            $endpoints[$endpointID]['errorred'] = true;
-            return null;
+            $status = curl_getinfo($curl_handle, CURLINFO_HTTP_CODE);
+            if ($status == 401) {
+                $errstr = "Authentication failed (error $status) while contacting $url. " .
+                    "Check credentials in restapi.secret.";
+                break;
+            } else if ($status != 200) {
```

Comment:
I think this should maybe any `2xx` code, and the same below for `5xx`.

Reply:
Fixed code to accept any 2xx code.
For 5xx, I think 500 error only matters. 500 is an error returned from DOMJudge, that implies there is an actual error in contest system. Other 5xx error usually occurs when the proxy in front of the DOMserver is busy.